### PR TITLE
[CALCITE-6220] Rewrite MIN/MAX(BOOLEAN) as BOOL_AND/BOOL_OR for Postgres, Redshift

### PR DESCRIPTION
**Pull request by tanclary**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/dialect/PostgresqlSqlDialect.java [hunk 1]
```diff
@@ -173,12 +173,16 @@ public PostgresqlSqlDialect(Context context) {
               timeUnitNode.getParserPosition());
       SqlFloorFunction.unparseDatetimeFunction(writer, call2, "DATE_TRUNC", false);
       break;
-
     default:
       super.unparseCall(writer, call, leftPrec, rightPrec);
     }
   }
 
+  @Override public SqlNode rewriteMaxMinExpr(SqlNode aggCall, RelDataType relDataType) {
```
Comment (julianhyde) [inline core/src/main/java/org/apache/calcite/sql/dialect/PostgresqlSqlDialect.java:181]: Don't instantatiate a dialect. Unnecessary coupling. Use a helper method. E.g. `static void unparseFetchUsingLimit()`.

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 2]
```diff
@@ -6677,12 +6677,37 @@ private void checkLiteral2(String expression, String expected) {
...
+    final String expectedPostgres = "SELECT BOOL_OR(\"brand_name\" = 'a'), "
+        + "BOOL_AND(\"brand_name\" = 'a'), "
+        + "MIN(\"brand_name\")\n"
+        + "FROM \"foodmart\".\"product\"";
+    final String expectedRedshift = "SELECT BOOL_OR(\"brand_name\" = 'a'), "
+        + "BOOL_AND(\"brand_name\" = 'a'), "
+        + "MIN(\"brand_name\")\n"
+        + "FROM \"foodmart\".\"product\"";
+    sql(query).ok(expected);
+    sql(query).withPostgresql().ok(expectedPostgres);
```
Comment (julianhyde) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:6707]: I would chain. And add a test for a 'neutral' dialect:
```
 sql(query).ok(expected)
    .withPostgresql().ok(expectedPostgres)
    .withRedshift().ok(expectedRedshift)
    .withMySQL().ok(expected);
```

I like how you tested on a non-BOOLEAN column

**Review by julianhyde — CHANGES_REQUESTED** [review]
I requested changes.

Also, in the jira summary, change `MIN/MAX(bool)` to `MIN/MAX(BOOLEAN)`. `bool` is not a SQL data type.

File: core/src/main/java/org/apache/calcite/sql/SqlDialect.java [hunk 3]
```diff
@@ -872,6 +872,13 @@ public SqlNode rewriteSingleValueExpr(SqlNode aggCall, RelDataType relDataType)
     return aggCall;
   }
 
+  /** With x as BOOLEAN column, rewrite MAX(x)/MIN(x) as BOOL_OR(x)/BOOL_AND(x)
```
[the remark this answers is not in the document]
Comment (tanclary) [reply core/src/main/java/org/apache/calcite/sql/SqlDialect.java]: Done

File: core/src/main/java/org/apache/calcite/sql/dialect/PostgresqlSqlDialect.java (hunk 1 above)
Comment (tanclary) [reply core/src/main/java/org/apache/calcite/sql/dialect/PostgresqlSqlDialect.java:181]: Done

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 2 above)
Comment (tanclary) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: Done

**Pull request merged**